utils/src/test/scala/com/salesforce/op/utils/stats/StreamingHistogramTest.scala

```diff
@@ -0,0 +1,262 @@
+    val grid2 = linspace(-1, 11, 25)
+    val grid3 = grid1 ++ grid2
+    val gaussian1 = Gaussian(0, 1)
+    val gaussian2 = Gaussian(5, 5)
+    val mixture = MixtureDistribution(gaussian1, gaussian2, 0.8)
+    val sampleSize = 500
+
+    cdfTest("Gaussian(0, 1)", gaussian1, grid1, sampleSize, 100, 50)
+    cdfTest("Gaussian(0, 1)", gaussian1, grid1, sampleSize, 200, 50)
+    cdfTest("Gaussian(0, 1)", gaussian1, grid1, sampleSize, 400, 50)
+    cdfTest("Gaussian(5, 5)", gaussian2, grid2, sampleSize, 100, 50)
+    cdfTest("Gaussian(5, 5)", gaussian2, grid2, sampleSize, 200, 50)
+    cdfTest("Gaussian(5, 5)", gaussian2, grid2, sampleSize, 400, 50)
+    cdfTest("0.8 * Gaussian(0, 1) + 0.2 * Gaussian(5, 5)", mixture, grid3, sampleSize, 100, 50)
+    cdfTest("0.8 * Gaussian(0, 1) + 0.2 * Gaussian(5, 5)", mixture, grid3, sampleSize, 200, 50)
+    cdfTest("0.8 * Gaussian(0, 1) + 0.2 * Gaussian(5, 5)", mixture, grid3, sampleSize, 400, 50)
+  }
+
+  it should "approximate Gamma distribution CDF" in {
+    val grid1 = linspace(0.0001, 4, 25)
+    val grid2 = linspace(950, 1050, 25)
+    val grid3 = grid1 ++ linspace(4, 950, 25) ++ grid2
+    val gamma1 = new Gamma(20, 1.0 / 2)
+    val gamma2 = new Gamma(1000000, 1.0 / 1000)
+    val mixture = MixtureDistribution(gamma1, gamma2, 0.95)
+    val sampleSize = 500
+
+    cdfTest("Gamma(20, 0.5)", gamma1, grid1, sampleSize, 100, 50)
+    cdfTest("Gamma(20, 0.5)", gamma1, grid1, sampleSize, 200, 50)
+    cdfTest("Gamma(20, 0.5)", gamma1, grid1, sampleSize, 400, 50)
+    cdfTest("Gamma(1000000, 0.001)", gamma2, grid2, sampleSize, 100, 50)
+    cdfTest("Gamma(1000000, 0.001)", gamma2, grid2, sampleSize, 200, 50)
+    cdfTest("Gamma(1000000, 0.001)", gamma2, grid2, sampleSize, 400, 50)
+    cdfTest("0.95 * Gamma(20, 0.5) + 0.05 * Gamma(1000000, 0.001)", mixture, grid3, sampleSize, 100, 50)
+    cdfTest("0.95 * Gamma(20, 0.5) + 0.05 * Gamma(1000000, 0.001)", mixture, grid3, sampleSize, 200, 50)
+    cdfTest("0.95 * Gamma(20, 0.5) + 0.05 * Gamma(1000000, 0.001)", mixture, grid3, sampleSize, 400, 50)
+  }
+
+  it should "approximate Beta distribution CDF" in {
+    val grid = linspace(0.0001, 0.999, 25)
+    val beta1 = new Beta(5, 1)
+    val beta2 = new Beta(1, 5)
+    val mixture = MixtureDistribution(beta1, beta2, 0.5)
+    val sampleSize = 500
+
+    cdfTest("Beta(5, 1)", beta1, grid, sampleSize, 100, 50)
+    cdfTest("Beta(5, 1)", beta1, grid, sampleSize, 200, 50)
+    cdfTest("Beta(5, 1)", beta1, grid, sampleSize, 400, 50)
+    cdfTest("Beta(1, 5)", beta2, grid, sampleSize, 100, 50)
+    cdfTest("Beta(1, 5)", beta2, grid, sampleSize, 200, 50)
+    cdfTest("Beta(1, 5)", beta2, grid, sampleSize, 400, 50)
+    cdfTest("0.5 * Beta(5, 1) + 0.5 * Beta(1, 5)", mixture, grid, sampleSize, 100, 50)
+    cdfTest("0.5 * Beta(5, 1) + 0.5 * Beta(1, 5)", mixture, grid, sampleSize, 200, 50)
+    cdfTest("0.5 * Beta(5, 1) + 0.5 * Beta(1, 5)", mixture, grid, sampleSize, 400, 50)
+  }
+
+  private def cdfTest(
```

Comment:
why there is so many private nested methods in this tests? it makes it very difficult to follow what actually the test does. can you please simplify if and use more of scalatest matchers syntax?